content/docs/intro/languages/javascript.md

```diff
@@ -31,6 +31,36 @@ $ pulumi new javascript
 
 This will create a `Pulumi.yaml` [project file]({{< relref "../concepts/project" >}}), a `package.json` file for dependencies, and an `index.js` file, containing your program. The name of the directory is used as the project name in `Pulumi.yaml`.
 
+## Optional `async` Entrypoint
```

Comment:
I think if we are going to document this here - we should probably broaden to describe the entrypoint more generally.

* The entrypoint file is specified in your package.json (`index.ts` by default)
* The entrypoint file is loaded as a Node.js module by Pulumi  (`require("index.ts")`)
* That module can do either:
   * Return a module object with properties for each stack output (`export const out = myResource.output`)
   * Set it's export to an async function whose return value is an object with members for each stack output (`export = async () => { }`)
* Most Pulumi programs use the first option, but programs that need to do `async` work at top level may find they want to use the second.

Reply:
Makes sense -- will do